src/main/java/com/redhat/devtools/lsp4ij/LanguageServiceAccessor.java

```diff
@@ -423,6 +423,26 @@ public CompletableFuture<Set<LanguageServerDefinition>> getAsyncMatched() {
         }
     }
 
+    /**
+     * Returns the language server wrappers for the specified file synchronously.
+     *
+     * @param file the file
+     * @return the language server wrappers that apply to the file
+     */
+    @ApiStatus.Internal
+    @NotNull
+    public Set<LanguageServerWrapper> getMatchedLanguageServerWrappersSync(@NotNull VirtualFile file) {

```

Comment:
If there's a better way to do this without adding this method, please let me know. It's needed to encapsulate finding external references by just a `PsiFile`, an offset, and a processor.